game_engine.py

```diff
@@ -172,13 +184,191 @@ def calculate_settlement_scores(self, player, current_board):
         return scores
     
     def calculate_development_score(self, player, current_board):
-        return 0
+        """
+        Calculate the score for choosing a development card this turn
+        :param player: The player object who is deciding what to do
+        :param current_board: The current state of the board
+        :return: the score * the weight based on the player strategy
+        """
+        score = 0
+        weight = 0
+        strategy = player.strategy
+        resources = player.resources
+
+        #  not enough resources
+        if resources["sheep"] == 0 or resources["stone"] == 0 or resources["wheat"] == 0:
+            return score
+
+        # enough resources, check strategy
+        if strategy == "development":
+            weight = 1
+        else:
+            weight = 0.65
+
+        # give score for not having enough knights for +2 points
+        if player.knights < 3:
+            score = score + 1
+
+        else:
```

Comment:
Shouldn't this be considered regardless of whether or not they have enough knights? If they need more nights and also there's a bunch of cards available then they should draw a development card.